### PR TITLE
Update Windwalker modules to TypeScript

### PR DESCRIPTION
**Pull request by Juko8**
Everything except resource tracking converted to TS

### CODE REVIEW

File: src/parser/monk/windwalker/modules/spells/TouchOfKarma.tsx [hunk 1]
```diff
@@ -3,29 +3,33 @@ import SPELLS from 'common/SPELLS';
...
-  constructor(options){
+  constructor(options: Options){
     super(options);
     this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.TOUCH_OF_KARMA_CAST), this.onCast);
   }
 
-  onCast(event) {
-    this.totalPossibleAbsorb += event.maxHitPoints * TOUCH_OF_KARMA_HP_SCALING;
+  onCast(event: CastEvent) {
+    this.totalPossibleAbsorb += event.maxHitPoints || 0 * TOUCH_OF_KARMA_HP_SCALING;
```
Comment (Dambroda) [inline src/parser/monk/windwalker/modules/spells/TouchOfKarma.tsx]: Parenthesis might be useful for reading here

File: src/parser/monk/windwalker/modules/Abilities.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ class Abilities extends CoreAbilities {
       {
         spell: SPELLS.FISTS_OF_FURY_CAST,
         category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
-        cooldown: haste => 24 / (1 + haste),
+        cooldown: (haste: number) => 24 / (1 + haste),
```
Comment (Dambroda) [inline src/parser/monk/windwalker/modules/Abilities.tsx]: `spellbook(): SpellbookAbility[] {`

prevents you needing to set `(haste: number)` in all of them :)

File: src/parser/monk/windwalker/modules/spells/SpinningCraneKick.tsx [hunk 3]
```diff
@@ -1,24 +1,36 @@
...
 import StatTracker from 'parser/shared/modules/StatTracker';
 import Statistic from 'interface/statistics/Statistic';
 import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText/index';
 import { STATISTIC_ORDER } from 'interface/others/StatisticBox';
-import Events from 'parser/core/Events';
+import Events, { ApplyDebuffEvent, CastEvent, RefreshDebuffEvent } from 'parser/core/Events';
 
-const isEqual = (a, b) => a.id === b.id && a.instance === b.instance
+interface MarkOfTheCrane {
+  target: MarkOfTheCraneTarget,
```
Comment (Dambroda) [inline src/parser/monk/windwalker/modules/spells/SpinningCraneKick.tsx]: style (also applies to the interface in combostrikes): semicolons and newlines (instance isnt on a newline in the interface below this)

**Review by Dambroda — APPROVED** [review]
a few minor optional style suggestions, otherwise LGTM

cant comment because there were no file changes, but in CONFIG.tsx you can make the following changes to add some more type checking:
```
import Config from 'parser/Config';
const config: Config = { .... }
export default config;
```

**Comment by Pewtro** [comment]
Closing this because I wasn't allowed to push to it to fix errors, sadge. Will merge in #4134, still accredited commits to you.

**Pull request closed without merging**